### ISSUE TITLE
message cannot be permanently deleted

### ISSUE THREAD

**Issue by dweorh** [amark/gun#1145]
I've found a strange behavior. I'm not sure if that's a bug or it works as designed but here is a code
`
let room = SEA.pair()
let user = SEA.pair()
let cert = await SEA.certify( 
  '*',
  { '*':'posts', '+': '*' },
  room
)

await gunRoom.get('posts') // => undefined

await gunUser.user().auth(user)
await gunUser.get('~' + room.pub).get('posts').get(user.pub).get('messages').put({message: 'hello'}, null, {opt: {cert: cert}})

gunRoom.user().get('posts').get(user.pub).get('messages') // => {message: hello}
gunRoom.user().get('posts').get(user.pub).put(null)

gunRoom.user().get('posts').get(user.pub).get('messages') // => undefined
gunRoom.user().get('posts').get(user.pub).set(null)

gunRoom.user().get('posts').get(user.pub).get('messages') // => {message: hello}
`

We can go a bit further with it

`
await gunUser.get('~' + room.pub).get('posts').get(user.pub).get('messages').put(null, null, {opt: {cert: cert}})
gunRoom.user().get('posts').get(user.pub).get('messages') // => undefined

await gunUser.get('~' + room.pub).get('posts').get(user.pub).get('messages').set(null, null, {opt: {cert: cert}})
gunRoom.user().get('posts').get(user.pub).get('messages') // => {message: hello, ku33iqysRfusABH: null}
`

When the owner of the message .set(null) it reverts the nullified by .put(null) document with some extra hash key with a null value.

Long story short. Once added message cannot be permanently deleted what causes issues like constantly growing graphs, and accidental restoration data.

@mimiza

**Comment by ruslanjan** [comment]
> and that's the problem because .get() of the nullified returns undefined but just after .set() the same .get() query returns more than we just added, as it reverts something from the past.
> 
> Imho, if .get() 'knows' that data was nullified, .set() should respect that as well

you can try set it {} or {deleted: true}. 

One way to prevent this is to design a graph the way it that would ditch unused data. You should avoid designing it the same way as with sql or document database. 

For example you can shard your data by months and if node is deleted then in the next month set it to null. and make users keep track of next month and current. then if user can't find data in this month node and Its not null then go to previous and look there if not there then node is too old and not relevant or deleted.

I know feels kinda bad, more like a workaround for a bug. But its graph database and you have to account to that.

**Comment by ruslanjan** [comment]
Currently I am not aware about how to make relay server not store some unused data or do cleanup but It should be possible
